From cfe424da5d5253ca2e6ef44058466b044b35274b Mon Sep 17 00:00:00 2001
From: Joaquim Rocha <me@joaquimrocha.com>
Date: Tue, 11 Sep 2018 00:15:02 +0200
Subject: [PATCH] Do not hardcode the assets location

This patch adds a config.h file the location for the shared assets and
modifies the files which previously had it hardcoded to include the
mentioned header.
---
 src/barragemanager.cc | 2 +-
 src/config.h          | 6 ++++++
 src/screen.c          | 2 +-
 src/soundmanager.c    | 2 +-
 4 files changed, 9 insertions(+), 3 deletions(-)
 create mode 100644 src/config.h

diff --git a/src/barragemanager.cc b/src/barragemanager.cc
index a96ab58..73536fc 100644
--- a/src/barragemanager.cc
+++ b/src/barragemanager.cc
@@ -18,11 +18,11 @@ extern "C" {
 }
 
 #include "barragemanager.h"
+#include "config.h"
 
 Barrage barragePattern[BARRAGE_TYPE_NUM][BARRAGE_PATTERN_MAX];
 int barragePatternNum[BARRAGE_TYPE_NUM];
 
-#define SHARE_LOC "/usr/share/games/rrootage/"
 //static const char *BARRAGE_DIR_NAME[BARRAGE_TYPE_NUM] = {
 //  "normal", "/usr/share/rRootage/reversible", "/usr/share/rRootage/morph", //"/usr/share/rRootage/simple", "/usr/share/rRootage/morph_heavy", //"/usr/share/rRootage/psy",
 //};
diff --git a/src/config.h b/src/config.h
new file mode 100644
index 0000000..b3c6599
--- /dev/null
+++ b/src/config.h
@@ -0,0 +1,6 @@
+#ifndef __CONFIG_H__
+#define __CONFIG_H__
+
+#define SHARE_LOC "/app/share/games/rRootage/"
+
+#endif /* __CONFIG_H__ */
diff --git a/src/screen.c b/src/screen.c
index 303ae49..f1a3830 100644
--- a/src/screen.c
+++ b/src/screen.c
@@ -24,6 +24,7 @@
 #include "attractmanager.h"
 #include "letterrender.h"
 #include "boss_mtd.h"
+#include "config.h"
 
 #define FAR_PLANE 720
 
@@ -31,7 +32,6 @@
 #define SCREEN_HEIGHT 480
 #define LOWRES_SCREEN_WIDTH 320
 #define LOWRES_SCREEN_HEIGHT 240
-#define SHARE_LOC "/usr/share/games/rrootage/"
 
 static Uint32 videoFlags;
 static int screenWidth, screenHeight;
diff --git a/src/soundmanager.c b/src/soundmanager.c
index f42c370..1de6492 100644
--- a/src/soundmanager.c
+++ b/src/soundmanager.c
@@ -17,11 +17,11 @@
 
 #include "SDL_mixer.h"
 #include "soundmanager.h"
+#include "config.h"
 
 static int useAudio = 0;
 
 #define MUSIC_NUM 3
-#define SHARE_LOC "/usr/share/games/rrootage/"
 
 static char *musicFileName[MUSIC_NUM] = {
   "stg_a.ogg", "stg_b.ogg", "stg_c.ogg",
-- 
2.17.1

